### PR TITLE
feat: add telegram token validation

### PR DESCRIPTION
**Pull request by Nikkov17**

### CODE REVIEW

File: src/core/fetchQuery.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  params,
+  ...options
+}: QueryOptions<RequestBody>): Promise<ResponseType> => {
+  const queryHeaders = new Headers({
+    authorization: ITIN_DATA,
+    ...headers,
+  });
+
+  let url = `${API_URL}${path}`;
+  if (params) url += `${new URLSearchParams(params)}${path}`;
```
Comment (AmsterGet) [inline src/core/fetchQuery.ts:33]: Why the search params specified before `path`?
Seems like here you've specified `path` twice.

File: src/core/fetchQuery.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { API_URL } from 'core/constants';
+
+// @ts-ignore
+// Telegram webapp object
+const tg = window.Telegram?.WebApp;
+
+const ITIN_DATA = tg.initData;
```
Comment (AmsterGet) [inline src/core/fetchQuery.ts:7]: `INIT_DATA`

Comment (Nikkov17) [reply src/core/fetchQuery.ts:7]: ))

File: src/core/fetchQuery.ts (hunk 1 above)
Comment (Nikkov17) [reply src/core/fetchQuery.ts:33]: yeah, good point, rewrote it several times, and it remains from different solutions, will fix it!

**Review by skyRoma — APPROVED** [review]

**Review by AmsterGet — APPROVED** [review]

**Pull request merged**